src/ica_api.c

```diff
@@ -1242,6 +1242,28 @@ int ica_ecdsa_verify(ica_adapter_handle_t adapter_handle,
 	return rc;
 }
 
+int ica_ec_key_get_public_key(ICA_EC_KEY *key, unsigned char *q, unsigned int *q_len)
```

Comment:
Can we improve the interface so that it's possible to call the function(s) first without
buffer to query for the buffer size:
int ica_ec_..._key(ICA_EC_KEY *key, unsigned char *q, unsigned int *q_len)
{
    ...
    if (q)
        memcpy(q, ...);
    *q_len = 2*privlen_from_nid(...);
    ...
}

Reply:
The current API is designed to return the private key's D-length when creating the key: 
ICA_EC_KEY* ica_ec_key_new(unsigned int nid, unsigned int *privlen);
So the application has this length when the key is created. 
We could improve the ica_ec_key_get_public_key() and ica_ec_key_get_private_key() functions to also provide this length, but it is not really necessary.